src/blob/index/core.rs

```diff
@@ -152,40 +180,43 @@ impl<FileIndex: FileIndexTrait> IndexStruct<FileIndex> {
             let size = findex.file_size() as usize;
             self.inner = State::OnDisk(findex);
             self.mem = None;
+            self.bloom_offset = bloom_offset;
             Ok(size)
         } else {
             Ok(0)
         }
     }
 
-    fn serialize_filters(&self) -> Result<Vec<u8>> {
+    fn serialize_filters(&self) -> Result<(Vec<u8>, usize)> {
         let range_buf = self.range_filter.to_raw()?;
         let range_buf_size = range_buf.len() as u64;
         let bloom_buf = self.bloom_filter.to_raw()?;
         let mut buf = Vec::with_capacity(size_of::<u64>() + range_buf.len() + bloom_buf.len());
+        let bloom_offset = size_of::<u64>() + range_buf.len();
         buf.extend_from_slice(&serialize(&range_buf_size)?);
         buf.extend_from_slice(&range_buf);
         buf.extend_from_slice(&bloom_buf);
-        Ok(buf)
+        Ok((buf, bloom_offset))
     }
 
-    fn deserialize_filters(buf: &[u8]) -> Result<(Bloom, RangeFilter)> {
+    fn deserialize_filters(buf: &[u8]) -> Result<(Bloom, RangeFilter, usize)> {
         let (range_size_buf, rest_buf) = buf.split_at(size_of::<u64>());
         let range_size = deserialize(&range_size_buf)?;
         let (range_buf, bloom_buf) = rest_buf.split_at(range_size);
         let bloom = Bloom::from_raw(bloom_buf)?;
         let range = RangeFilter::from_raw(range_buf)?;
-        Ok((bloom, range))
+        Ok((bloom, range, range_size + size_of::<u64>()))
```

Comment:
I think it would be nice to add the `offset` field directly to the `Bloom` structure. By default it has a `None` value, but when the filter is dumped to disk or created from disk, the value in this field updated to be the actual offset. This approach allow to remove that additional element in the tuple

Reply:
Done